src/main/java/de/dennisguse/opentracks/views/IntervalListView.java

```diff
@@ -0,0 +1,96 @@
+package de.dennisguse.opentracks.views;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.LinearLayout;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import java.util.List;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.adapters.IntervalStatisticsAdapter;
+import de.dennisguse.opentracks.util.PreferencesUtils;
+import de.dennisguse.opentracks.viewmodels.IntervalStatistics;
+import de.dennisguse.opentracks.viewmodels.IntervalStatisticsModel;
+
+/**
+ * LinearLayout view used to build a list of intervals.
+ * See {@link IntervalStatisticsAdapter}.
+ */
+public class IntervalListView extends LinearLayout {
+
+    protected IntervalStatisticsAdapter adapter;
+    protected LinearLayout linearLayoutIntervals;
+    protected Spinner spinnerIntervals;
+    protected TextView spinnerIntervalsUnit;
+
+    protected Context context;
+    protected IntervalListListener listener;
+
+    protected final SharedPreferences.OnSharedPreferenceChangeListener sharedPreferenceChangeListener = (preferences, key) -> {
+        if (PreferencesUtils.isKey(getContext(), R.string.stats_units_key, key) || PreferencesUtils.isKey(getContext(), R.string.stats_rate_key, key)) {
+            if (spinnerIntervalsUnit != null) {
+                spinnerIntervalsUnit.setText(PreferencesUtils.isMetricUnits(context) ? context.getString(R.string.unit_kilometer) : context.getString(R.string.unit_mile));
+            }
+        }
+    };
+
+    public IntervalListView(Context context, IntervalListListener listener) {
+        super(context);
+        this.context = context;
```

Comment:
Can we not use the content of the parent?
(PS: don't know)

Reply:
Yes, we can use `getContext()`. Done.